server/src/utils/brute/removeDestinyChoiceFromDestinyPath.ts

```diff
@@ -0,0 +1,71 @@
+import { randomBetween } from '@labrute/core';
+import {
+  Brute,
+  BruteStat,
+  DestinyChoiceType,
+  PetName,
+  PrismaClient,
+  SkillName,
+  WeaponName,
+} from '@labrute/prisma';
+
+export const removeDestinyChoiceFromDestinyPath = async (
+  prisma: PrismaClient,
+  brute: Pick<Brute, 'id'>,
+  destinyChoice: WeaponName | SkillName | PetName,
+) => {
+  const destinyChoices = await prisma.destinyChoice.findMany({
+    where: {
+      bruteId: brute.id,
+    },
+  });
+
+  const isWeapon = Object.values(WeaponName).includes(destinyChoice as WeaponName);
+  const isSkill = Object.values(SkillName).includes(destinyChoice as SkillName);
+  const isPet = Object.values(PetName).includes(destinyChoice as PetName);
+
+  if (!isWeapon && !isSkill && !isPet) {
+    return;
+  }
+
+  // Replace the destiny choice in destiny path with random stats
+  const bruteStats = Object.values(BruteStat);
+
+  for (const choice of destinyChoices) {
+    if (isWeapon && (choice.type !== DestinyChoiceType.weapon || choice.weapon !== destinyChoice)) {
+      continue;
+    }
+    if (isSkill && (choice.type !== DestinyChoiceType.skill || choice.skill !== destinyChoice)) {
+      continue;
+    }
+    if (isPet && (choice.type !== DestinyChoiceType.pet || choice.pet !== destinyChoice)) {
+      continue;
+    }
```

Comment:
Instead of querying the whole destiny, query only the choices that correspond to the current perk to change. So set isWeapon/isSkill/isPet before querying and use that to refine the query and loop on the results after